src/App.js

```diff
@@ -1,113 +1,122 @@
-
-  const [status,setStatus] = useState("all")
+  const [status, setStatus] = useState("all")
 
   const [filteredTodos, setFilteredTodos] = useState([])
-   const [inputName, setInputName] = useState("");
-   const [inputpassword, setinputpassword] = useState("");
+  const [inputName, setInputName] = useState("")
+  const [inputpassword, setinputpassword] = useState("")
 
   const [isAdmin, setIsAdmin] = useState(false)
   const [redirect, setRedirect] = useState(false)
-  
-  
-  
-
-   const loginhandler = (e) => {
-     e.preventDefault();
-     if (inputName === "asim" && inputpassword === "asim") {
-       //redirected to App.js
+  const [alert, setAlert] = useState(false)
+
+  const loginhandler = (e) => {
+    e.preventDefault()
+    if (inputName === "asim" && inputpassword === "asim") {
+      //redirected to App.js
       setIsAdmin(false)
       setRedirect(true)
-       
+      setAlert(false)
+
       //  return <Redirect to="/todo" true/>;
-     }
-     else if (inputName === "admin" && inputpassword === "admin") {
-       
-       setIsAdmin(true);
-       setRedirect(true)
+    } else if (inputName === "admin" && inputpassword === "admin") {
+      setIsAdmin(true)
+      setRedirect(true)
+      setAlert(false)
       //  return <Redirect to="/todo" true/>
-     }
-     else {
-       alert("Invalid Credentials")
-     }
-     
-   };
+    } else {
+      setAlert(true)
+    }
+  }
 
-   const nameChangeHandler = (e) => {
-     setInputName(e.target.value);
-   };
+  const nameChangeHandler = (e) => {
+    setInputName(e.target.value)
+  }
 
-   const passwordChangeHandler = (e) => {
-     setinputpassword(e.target.value);
-   };
+  const passwordChangeHandler = (e) => {
+    setinputpassword(e.target.value)
+  }
 
-  useEffect(()=>{getLocalTodos()},[])
+  useEffect(() => {
+    getLocalTodos()
+  }, [])
 
   useEffect(() => {
     filterHandler()
-    saveLocalTodos()}, [todos, status]);
-  
+    saveLocalTodos()
+  }, [todos, status])
+
   const filterHandler = () => {
     switch (status) {
       case "completed":
-        setFilteredTodos(todos.filter(todo => todo.completed === true))
-        break;
+        setFilteredTodos(todos.filter((todo) => todo.completed === true))
+        break
       case "incompleted":
-        setFilteredTodos(todos.filter(todo => todo.completed === false))
-        break;
+        setFilteredTodos(todos.filter((todo) => todo.completed === false))
+        break
       default:
         setFilteredTodos(todos)
-        break;
+        break
     }
   }
 
-//save to localStorage
-  
+  //save to localStorage
+
   const saveLocalTodos = () => {
-      localStorage.setItem("todos",JSON.stringify(todos))
+    localStorage.setItem("todos", JSON.stringify(todos))
   }
 
   //get local todos
 
   const getLocalTodos = () => {
     if (localStorage.getItem("todos") === null) {
-      localStorage.setItem("todos", JSON.stringify([]));
+      localStorage.setItem("todos", JSON.stringify([]))
```

Comment:
Why are we saving empty array in local storage if null?

Reply:
I did that while testing. Removed that condition to check get todo array if it exists only